src/auth_storage.js

```diff
@@ -0,0 +1,98 @@
+export class LocalStorage {
+  constructor (prefix = 'cozy:oauth:', getLocalStorage = () => window.localStorage) {
+    // We try to deref the localstorage, and if it fails (because the user has
+    // deactivated its localstorage for instance) we fallback on a memory
+    // storage.
```

Comment:
I don't think it's a good idea to silently fallback to memstorage if localStorage can't be used. The typical use case for localStorage is a SPA, and memStorage won't work in that case because the page will be unloaded and reloaded later.